sw/device/tock/chips/earlgrey/src/plic.rs

```diff
@@ -0,0 +1,95 @@
+//! Platform Level Interrupt Control peripheral driver.
+
+use tock::kernel as kernel;
+
+use kernel::common::registers::{register_bitfields, register_structs, ReadOnly, ReadWrite};
+use kernel::common::StaticRef;
+
+register_structs! {
+    pub PlicRegisters {
+        /// Interrupt Pending Register
+        (0x000 => pending: [ReadOnly<u32>; 3]),
+        /// Interrupt Source Register
+        (0x00C => source: [ReadWrite<u32>; 3]),
+        /// Interrupt Priority Registers
+        (0x018 => priority: [ReadWrite<u32, priority::Register>; 79]),
+        (0x154 => _reserved0: [ReadWrite<u32>; 43]),
+        /// Interrupt Enable Register
+        (0x200 => enable: [ReadWrite<u32>; 3]),
+        /// Priority Threshold Register
+        (0x20C => threshold: ReadWrite<u32, priority::Register>),
+        /// Claim/Complete Register
+        (0x210 => claim: ReadWrite<u32>),
+        /// MSIP Register
+        (0x214 => msip: ReadWrite<u32>),
+        (0x218 => @END),
+    }
+}
+
+register_bitfields![u32,
+    priority [
+        Priority OFFSET(0) NUMBITS(3) []
+    ]
+];
+
+const PLIC_BASE: StaticRef<PlicRegisters> =
+    unsafe { StaticRef::new(0x4009_0000 as *const PlicRegisters) };
+
+/// Clear all pending interrupts.
+pub unsafe fn clear_all_pending() {
+    let _plic: &PlicRegisters = &*PLIC_BASE;
```

Comment:
It's not clear what this is doing. Why are you creating a binding and not using it? Is there code missing. or dos the deref operation have a side-effect?